engine-types/src/parameters.rs

```diff
@@ -64,3 +81,18 @@ pub struct RefundCallArgs {
     pub erc20_address: Option<Address>,
     pub amount: RawU256,
 }
+
+/// Args passed to the the cross contract call precompile.
+/// That precompile is used by Aurora contracts to make calls to the broader NEAR ecosystem.
+/// See https://github.com/aurora-is-near/AIPs/pull/2 for design details.
+#[derive(Debug, BorshSerialize, BorshDeserialize)]
+pub enum CrossContractCallArgs {
+    /// The promise is to be executed immediately (as part of the same NEAR transaction as the EVM call).
+    Eager(PromiseArgs),
+    /// The promise is to be stored in the router contract, and can be executed in a future transaction.
+    /// The purpose of this is to expand how much NEAR gas can be made available to a cross contract call.
+    /// For example, if an expensive EVM call ends with a NEAR cross contract call, then there may not be
+    /// much gas left to perform it. In this case, the promise could be `Delayed` (stored in the router)
+    /// and executed in a separate transaction with a fresh 300 Tgas available for it.
+    Delayed(PromiseArgs),
```

Comment:
What if these are never consumed? Could we have some process to free these up if they are old? Or should we?

Reply:
Ultimately it is the user that has to cover the storage staking for stored promises they never used. I don't know why they would not use the promises they have scheduled. We could add a cancellation interface, but this adds more complexity to the design, and I do not think it is worth having that complexity for this edge case that may never come up.

Reply:
Also this bring the question: who can destroy it and when? probably the answer is the same user that scheduled it, but is still a bit pointless! Agree on leaving it there until it is consumed!